src/game/MapManager.cpp

```diff
@@ -172,8 +175,15 @@ void MapManager::Update(uint32 diff)
     if (!i_timer.Passed())
         return;
 
-    for (MapMapType::iterator iter = i_maps.begin(); iter != i_maps.end(); ++iter)
-        iter->second->Update((uint32)i_timer.GetCurrent());
+	for (MapMapType::iterator iter = i_maps.begin(); iter != i_maps.end(); ++iter)
+	{
+		if(m_updater.activated())
+			m_updater.schedule_update(*iter->second, uint32(i_timer.GetCurrent()));
+		else
+			iter->second->Update((uint32)i_timer.GetCurrent());
+	}
+	if (m_updater.activated())
+		m_updater.wait();
```

Comment:
Each tick expected to be global. So only tick proccessing in parallel for map instances.